File: mobile/lib/core/models/program_models.dart
class ProgramResponse {
  final int id;
  final String code;
  final String name;
  final String? description;
  final bool isActive;

  const ProgramResponse({
    required this.id,
    required this.code,
    required this.name,
    this.description,
    required this.isActive,
  });

  factory ProgramResponse.fromJson(Map<String, dynamic> json) => ProgramResponse(
        id: json['id'] as int,
        code: json['code'] as String,
        name: json['name'] as String,
        description: json['description'] as String?,
        isActive: json['isActive'] as bool,
      );
}

File: web/src/app/core/models/program.models.ts
export interface ProgramResponse {
  id: number;
  code: string;
  name: string;
  description: string | null;
  isActive: boolean;
  createdAt: string;
}

export interface CreateProgramRequest {
  code: string;
  name: string;
  description?: string;
}

export interface UpdateProgramRequest {
  name: string;
  description?: string;
  isActive: boolean;
}

export interface UserProgramResponse {
  programId: number;
  code: string;
  name: string;
  grantedAt: string;
  grantedByUsername: string | null;
}
